modules/onlp_snmp/module/src/onlp_snmp_sensors.c

```diff
@@ -34,113 +34,214 @@
 #include <onlp/thermal.h>
 #include <onlp/fan.h>
 #include <onlp/psu.h>
-#include <onlp/led.h>
 
 #include "onlp_snmp_log.h"
 
+
 /**
  * Individual Sensor Control structure.
  */
 typedef struct onlp_snmp_sensor_s {
-    int sensor_id; /* sensor identification */
+    list_links_t links;  /* for tracking sensors of the same type */
+    int sensor_id;       /* onlp_oid_t for this sensor */
     char name[ONLP_SNMP_CONFIG_MAX_NAME_LENGTH];
     char desc[ONLP_SNMP_CONFIG_MAX_DESC_LENGTH];
-    int sensor_type;
+    onlp_snmp_sensor_type_t sensor_type;
+    uint32_t index;      /* snmp table column */
     union sensor_info {
         onlp_thermal_info_t ti;
         onlp_fan_info_t     fi;
         onlp_psu_info_t     pi;
-        onlp_led_info_t     li;
-        uint32_t            mi; /* this is for misc value */
     } sensor_info;
-    bool info_valid;             /* true if sensor_info is valid */
-    uint64_t last_update_time;   /* last time called */
+    /* for snmp table maintenance:
```

Comment:
in collect_sensors__, we assign the index to ONLP_OID_ID_GET(oid).  so if the onlp oid stays fixed for a given sensor, then the sensor index should also be fixed.